MastermindLogic/src/main/java/nl/fhict/s3/mastermindlogic/GameEndpoint.java

```diff
@@ -111,19 +119,28 @@ private void cannotParseMessage(String message) {
     }
 
     private void registerGameOperation(Session session) {
-        Game game = new Game();
-        application.newGame(game);
+        Game game = application.getOpenGameOrNew();
+        int playerId = -1;
+        if(game.getPlayer1() == null) {
+            game.setPlayer1(new Player(0));
+            playerId = 0;
+        } else if(game.getPlayer2() == null) {
+            game.setPlayer2(new Player(1));
+            playerId = 1;
+        } else {
+            //TODO: error
+        }
+        
+        assert playerId != -1;
+
         games.put(game.getId(), new ArrayList<Session>());
         games.get(game.getId()).add(session);
         
         // Return gameID to user
-        Gson gson = new Gson();
-        WebSocketMessage returnMessage = new WebSocketMessage();
-        returnMessage.setGameId(game.getId());
-        returnMessage.setOperation(WebSocketMessageOperation.REGISTER_GAME);
-        returnMessage.setContent(gson.toJson(game.getId().toString()));
-        String jsonReturnMessage = gson.toJson(returnMessage);
-        session.getAsyncRemote().sendText(jsonReturnMessage);
+        sendMessage(session, playerId, game.getId(), WebSocketMessageOperation.REGISTER_GAME, game.getId().toString());
+        
+        // return playerID to user
+        sendMessage(session, playerId, game.getId(), WebSocketMessageOperation.JOIN_GAME, Integer.toString(playerId));
```

Comment:
kan ook gewoon in registerGame playerId changen

Reply:
Heb ik gefixed